core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,150 @@
 package webapi
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
 	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/workflow"
 	"github.com/smartcontractkit/chainlink/v2/core/services/job"
 )
 
+const defaultSendChannelBufferSize = 1000
+
+type Response struct {
+	Success      bool   `json:"success"`
+	ErrorMessage string `json:"error_message,omitempty"`
+}
+
+type workflowConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	connector connector.GatewayConnector
+	lggr      logger.Logger
+	mu        sync.Mutex
+	triggers  map[string]chan capabilities.TriggerResponse
+}
+
+var _ capabilities.TriggerCapability = (*workflowConnectorHandler)(nil)
+var _ services.Service = &workflowConnectorHandler{}
+
 func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (job.ServiceCtx, error) {
 	// TODO (CAPPL-22, CAPPL-24):
 	//   - decode config
 	//   - create an implementation of the capability API and add it to the Registry
 	//   - create a handler and register it with Gateway Connector
 	//   - manage trigger subscriptions
 	//   - process incoming trigger events and related metadata
-	return nil, nil
+
+	handler := &workflowConnectorHandler{
+		connector: connector,
+		lggr:      lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	// is this the right way to register with gateway connector?  Cron trigger doesn't do this.
```

Comment:
Cron trigger doesn't have connectors/handlers.

What I'd consider here is should this be in `.Start` instead? Trace through where constructor gets called versus where `.Start` gets called - is the connector ready for handlers at that point?